File: frontend/mobile/lib/models/festival.dart
class Festival {
  const Festival({
    required this.id,
    required this.name,
    required this.description,
    required this.startDate,
    required this.endDate,
    required this.image,
    required this.organiserName,
    required this.organiserProfilePicture,
  });

  final int id;
  final String name;
  final String description;
  final DateTime startDate;
  final DateTime endDate;
  final String? image;
  final String organiserName;
  final String? organiserProfilePicture;

  factory Festival.fromJson(
    Map<String, dynamic> json,
  ) {
    return Festival(
      id: json['id'] as int,
      name:
          json['name'] as String? ??
          'Unnamed festival',
      description:
          json['description'] as String? ??
          '',
      startDate: DateTime.parse(
        json['start_date'] as String,
      ),
      endDate: DateTime.parse(
        json['end_date'] as String,
      ),
      image: json['image'] as String?,
      organiserName:
          json['organiser_name']
              as String? ??
          '',
      organiserProfilePicture:
          json['organiser_profile_picture']
              as String?,
    );
  }
}

File: frontend/browser/src/types/festival.ts
export type Festival = {
    id: number;
    name: string;
    description: string;
    start_date: string;
    end_date: string;
    organiser: number;
    image: string | null;
    created_at: string;
    updated_at: string;
};

export type Stage = {
    id: number;
    festival: number;
    name: string;
};

export type FestivalSet = {
    id: number;

    stage: number;
    stage_name: string;

    artist: number;
    artist_name: string;

    start_time: string;
    end_time: string;

    image: string | null;

    is_liked: boolean;
    like_count: number;
};

export type CreateStageRequest = {
    name: string;
};

export type CreateSetRequest = {
    stage: number;
    artist: string;
    start_time: string;
    end_time: string;
};

export type Artist = {
    id: number;
    festival: number;
    name: string;
    description: string;
    image: string | null;
    created_at: string;
    updated_at: string;
};

export type PublicFestival = {
    id: number;

    name: string;
    description: string;

    start_date: string;
    end_date: string;

    image: string | null;

    organiser_name: string;
    organiser_profile_picture: string | null;
};